packages/eslint-plugin/rules/i18n-no-flanking-whitespace.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Internal dependencies
+ */
+const {
+	TRANSLATION_FUNCTIONS,
+	getTextContentFromNode,
+	getTranslateFunctionName,
+	getTranslateFunctionArgs,
+} = require( '../utils' );
+
+const PROBLEMS_BY_CHAR_CODE = {
+	9: '\\t',
+	10: '\\n',
+	13: '\\r',
+	32: 'whitespace',
+};
+
+function makeFixerFunction( arg ) {
+	return ( fixer ) => {
+		switch ( arg.type ) {
+			case 'Literal':
+				return [ fixer.replaceText( arg, `'${ arg.value.trim() }'` ) ];
+
+			case 'BinaryExpression':
+				return [
+					...makeFixerFunction( arg.left )( fixer ),
+					...makeFixerFunction( arg.right )( fixer ),
+				];
```

Comment:
Good catch! Again fixing this fixer might get complicated with concatenated strings. I'm thinking the simplest thing is to have the fixer only apply to the simplest case. Thoughts?  

Hmm I'll have to check to see if `__(  'This is a ' + 'valid string' )` fails since the trailing whitespace in the first string segment should be valid. Joining strings in the `__` function doesn't not seem like a good idea but something like that might trip up the lint rule.